examples/7_qst_circuit.jl

```diff
@@ -29,11 +31,11 @@ N = length(Ψ)     # Number of qubits
 ψ0 = randomstate(Ψ; χ=χ)
 
 # Initialize stochastic gradient descent optimizer
-opt = SGD(; η=0.01)
+opt = Optimisers.Descent(0.01)
 
 # Initialize the observer for the fidelity
-F(ψ::MPS) = fidelity(ψ, Ψ)
-obs = Observer(F)
+F(ψ::MPS; kwargs...) = fidelity(ψ, Ψ)
```

Comment:
So here in principle the wavefunction `ψ` could have been passed as a keyword argument, right?

I guess in general we need to have a specification for each function about which values should get passed as regular arguments vs. keyword arguments. Is the idea that we will pass what we think are the most commonly used values as arguments to the measurement functions?

Reply:
No you are right here. We should always use only kwargs. I will rectify this.

Reply:
Also I wish we had a better way to do it besides forcing people to define the function with `F(ψ::MPS; kwargs...) = fidelity(ψ, Ψ)` instead of just `F(ψ::MPS) = fidelity(ψ, Ψ)`.

Actually, we may be able to do it with [`hasmethod`](https://docs.julialang.org/en/v1/base/base/#Base.hasmethod) (we could check if the function passed accepts keyword arguments, and if not call it without keyword arguments). But that could be left for future work.